### ISSUE TITLE
Add support for Twitter OAuth1a "x_auth_access_type"

### ISSUE THREAD

**Issue by chez14** [simov/grant#272]
Documentation Reference: https://developer.twitter.com/en/docs/authentication/api-reference/request_token

Based on the documentation, we can customize the login scope of the app. The scope definition must be provided when requesting token via `x_auth_access_type`.

Reviewing the grant's [OAuth1 flow](https://github.com/simov/grant/blob/master/lib/flow/oauth1.js) code, we can either use `providers.scope`, or we can add `providers.custom-param.x_auth_access_type` to fill the `x_auth_access_type` param.

Thank you.

**Comment by simov** [comment]
Hi, I pushed it here https://github.com/simov/grant/commit/c6b76b78e3ee36358c74ee7bc2155dac945d3a97 but I need to add some tests for it.

In the meantime you can reference it as `github:simov/grant` in your package.json and use it with either one of the two:

```json
"twitter": {
  "custom_params": {"x_auth_access_type": "read"}
}
"twitter": {
  "scope": ["read"]
}
```

**Comment by chez14** [comment]
Thank you so much! It works at my end.

| With `write` scope | With `read` scope |
|-------------------------|--------------------------|
| ![image](https://user-images.githubusercontent.com/5555026/156946804-70362282-145e-40eb-8d99-bb8eacce7903.png) | ![image](https://user-images.githubusercontent.com/5555026/156946812-9644a1d3-2e42-41ea-a607-657124a3880e.png) |

**Comment by simov** [comment]
Published in v5.4.21
